src/launchpad/size/models/apple.py

```diff
@@ -72,32 +71,67 @@ class AppleAppInfo(BaseAppInfo):
     main_binary_uuid: str | None = Field(None, description="UUID of the main binary")
 
 
-class MachOBinaryAnalysis(BaseBinaryAnalysis):
-    """Mach-O binary analysis results."""
+@dataclass
+class SegmentInfo:
```

Comment:
I created types to capture some of the `lief` info we were having issues with memory leaks using data after the file is closed.